api/v1alpha1/release_types.go

```diff
@@ -38,95 +38,71 @@ type ReleaseSpec struct {
 )
 
 // ReleaseStatus defines the observed state of Release.
 type ReleaseStatus struct {
-	// StartTime is the time when the Release PipelineRun was created and set to run
+	// Conditions represent the latest available observations for the release
 	// +optional
-	StartTime *metav1.Time `json:"startTime,omitempty"`
+	Conditions []metav1.Condition `json:"conditions"`
 
-	// CompletionTime is the time the Release PipelineRun completed
 	// +optional
-	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+	Deployment DeploymentInfo `json:"deployment,omitempty"`
 
-	// DeploymentStartTime is the time when the SnapshotEnvironmentBinding was created
 	// +optional
-	DeploymentStartTime *metav1.Time `json:"deploymentStartTime,omitempty"`
+	Release ReleaseInfo `json:"release,omitempty"`
 
-	// DeploymentCompletionTime is the time when the SnapshotEnvironmentBinding has all components deployed
+	// Target references where this release is intended to be released to
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	// +optional
-	DeploymentCompletionTime *metav1.Time `json:"deploymentCompletionTime,omitempty"`
+	Target string `json:"target,omitempty"`
+}
 
-	// Conditions represent the latest available observations for the release
+type DeploymentInfo struct {
+	// CompletionTime is the time the SnapshotEnvironmentBinding has all components deployed
 	// +optional
-	Conditions []metav1.Condition `json:"conditions"`
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
 
 	// SnapshotEnvironmentBinding contains the namespaced name of the SnapshotEnvironmentBinding created as part of
 	// this release
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?\/[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	// +optional
 	SnapshotEnvironmentBinding string `json:"snapshotEnvironmentBinding,omitempty"`
 
-	// ReleasePipelineRun contains the namespaced name of the release PipelineRun executed as part of this release
+	// StartTime is the time when the SnapshotEnvironmentBinding was created
+	// +optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+}
+
+type ReleaseInfo struct {
+	// CompletionTime is the time the Release PipelineRun completed
+	// +optional
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// PipelineRun contains the namespaced name of the release PipelineRun executed as part of this release
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?\/[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	// +optional
-	ReleasePipelineRun string `json:"releasePipelineRun,omitempty"`
+	PipelineRun string `json:"pipelineRun,omitempty"`
 
-	// ReleaseStrategy contains the namespaced name of the ReleaseStrategy used for this release
+	// Strategy contains the namespaced name of the Strategy used for this release
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?\/[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	// +optional
-	ReleaseStrategy string `json:"releaseStrategy,omitempty"`
```

Comment:
I prefer ReleaseStrategy

Reply:
In the yaml would appear as:
```
Release:
    ReleaseStrategy:
```
Same goes for code references:
```
release.Status.Release.ReleaseStrategy
```
instead of
```
release.Status.Release.Strategy
```
Not a fan. Exactly the same applies to Go packages and struct names. They "forbid" it.

Reply:
I kind of prefer keeping the things in `Status.Release` toplevel. I do not like going to `Strategy`. A tekton task has a `taskref` not just `ref`. I don't think we should be calling things this short name. If you really want it grouped, maybe pick a name for this key not called `Release`

Reply:
Didn't we already pick process for this?